### PR TITLE
[Flax] Refactor gpt2 & bert example docs

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR mainly refactors the docs of the official Flax MLM, CLM examples. The CLM training script is also slightly changed for consistency with the MLM script.

### CODE REVIEW

File: examples/flax/language-modeling/README.md [hunk 1]
```diff
@@ -180,25 +180,51 @@ Next we clone the model repository to add the tokenizer and model files.
 git clone https://huggingface.co/<your-username>/norwegian-gpt2
 ```
 
-To ensure that all tensorboard traces will be uploaded correctly, we need to 
-track them. You can run the following command inside your model repo to do so.
+To setup all relevant files for trairing, let's go into the cloned model directory.
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/README.md:183]:
```suggestion
To setup all relevant files for training, let's go into the cloned model directory.
```

File: examples/flax/language-modeling/run_clm_flax.py [hunk 2]
```diff
@@ -182,18 +184,16 @@ def data_loader(rng: jax.random.PRNGKey, dataset: Dataset, batch_size: int, shuf
     steps_per_epoch = len(dataset) // batch_size
 
     if shuffle:
-        batch_idx = jax.random.permutation(rng, len(dataset))
+        batch_idx = np.random.permutation(len(dataset))
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/run_clm_flax.py:187]: @patil-suraj - those changes here also seem to give ca. a 2x speed-up of the overall training. Will run on the full example now to verify.

Comment (patrickvonplaten) [reply examples/flax/language-modeling/run_clm_flax.py:187]: Hmm weird on a subset of the dataset training is much faster then on the whole dataset. Training on the whole dataset doesn't really give a speed-up...

Comment (patrickvonplaten) [reply examples/flax/language-modeling/run_clm_flax.py:187]: Performance seems to be identical more or less...leaving it this way so for consistency with the MLM and T5-MLM script

File: examples/flax/language-modeling/README.md [hunk 3]
```diff
@@ -49,37 +49,29 @@ Next we clone the model repository to add the tokenizer and model files.
 git clone https://huggingface.co/<your-username>/norwegian-roberta-base
 ```
 
-To ensure that all tensorboard traces will be uploaded correctly, we need to 
-track them. You can run the following command inside your model repo to do so.
+To setup all relevant files for trairing, let's go into the cloned model directory.
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/README.md:52]:
```suggestion
To setup all relevant files for training, let's go into the cloned model directory.
```

**Review by sgugger — APPROVED** [review]
LGTM, thanks for refactoring!

**Review by patil-suraj — APPROVED** [review]
LGTM!
